RockWeb/Blocks/Core/MergeTemplateEntry.ascx.cs

```diff
@@ -140,6 +147,7 @@ protected void btnMerge_Click( object sender, EventArgs e )
             // NOTE: This is a full postback (not a partial like most other blocks)
 
             var rockContext = new RockContext();
+            rockContext.Database.CommandTimeout = 600;
```

Comment:
this could be a block setting,  for example https://github.com/SparkDevNetwork/Rock/blob/d94bfb29f4b376d6596bf1d06a285ff29c208863/RockWeb/Blocks/Reporting/ReportDetail.ascx.cs#L42